pkg/cmd/util/clientcmd/clientcmd.go

```diff
@@ -76,6 +87,24 @@ func EnvVars(host string, caData []byte, insecure bool, bearerTokenFile string)
 func (cfg *Config) bindEnv() error {
 	var err error
 
+	if cfg.clientConfig != nil {
+		clientConfig, err := cfg.clientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		cfg.CommonConfig = *clientConfig
+		cfg.Namespace, _, err = cfg.clientConfig.Namespace()
+		if err != nil {
+			return err
+		}
+		if !cfg.MasterAddr.Provided {
+			cfg.MasterAddr.Set(cfg.CommonConfig.Host)
+		}
+		if !cfg.KubernetesAddr.Provided {
+			cfg.KubernetesAddr.Set(cfg.CommonConfig.Host)
+		}
+	}
```

Comment:
So the intent here is that the environment variables will trump the kubeconfig value?  That seems a little odd.